### PR TITLE
Address layering of navigation action in code actions

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Features/Core/Portable/Common/DocumentNavigationOperation.cs [hunk 1]
```diff
@@ -21,25 +19,17 @@ namespace Microsoft.CodeAnalysis.CodeActions
...
+            DocumentId = documentId;
+            Position = position;
         }
 
         public override void Apply(Workspace workspace, CancellationToken cancellationToken)
         {
-            if (workspace.CanOpenDocuments)
-            {
-                var navigationService = workspace.Services.GetService<IDocumentNavigationService>();
-                var threadingService = workspace.Services.GetService<IWorkspaceThreadingServiceProvider>();
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/Common/DocumentNavigationOperation.cs:38]: Can you delete IWorkspaceThreadingServiceProvider or did something else still use it too?

**Review by jasonmalinowski — APPROVED** [review]
Nice approach; didn't realize we also had the navigation annotation too so makes sense all navigation is done in one place anyways.

File: src/Features/Core/Portable/Common/DocumentNavigationOperation.cs [hunk 2]
```diff
@@ -21,25 +19,17 @@ namespace Microsoft.CodeAnalysis.CodeActions
...
+        public DocumentNavigationOperation(DocumentId documentId!!, int position = 0)
         {
-            _documentId = documentId ?? throw new ArgumentNullException(nameof(documentId));
-            _position = position;
+            DocumentId = documentId;
+            Position = position;
         }
 
         public override void Apply(Workspace workspace, CancellationToken cancellationToken)
         {
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/Common/DocumentNavigationOperation.cs:32]: true.

File: src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs [hunk 3]
```diff
@@ -289,41 +288,50 @@ internal class CodeActionEditHandlerService : ForegroundThreadAffinitizedObject,
...
 
-        private async Task TryNavigateToLocationOrStartRenameSessionAsync(Workspace workspace, Solution oldSolution, Solution newSolution, CancellationToken cancellationToken)
+        private async Task TryNavigateToLocationOrStartRenameSessionAsync(
+            Workspace workspace,
+            ImmutableArray<CodeActionOperation> operations,
+            Solution oldSolution,
+            Solution newSolution,
+            CancellationToken cancellationToken)
         {
+            var navigationOperation = operations.OfType<DocumentNavigationOperation>().FirstOrDefault();
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs:298]: well SingleOrDefault throws... and that worries me about somethign potentially throwing now taht just had wonky behavior before.

File: src/Features/Core/Portable/Common/DocumentNavigationOperation.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/Common/DocumentNavigationOperation.cs:38]: unfortunately, unit testing and TS are still using it in exactly 2 locations.

File: src/Features/Core/Portable/Common/DocumentNavigationOperation.cs [hunk 4]
```diff
@@ -21,25 +19,17 @@ namespace Microsoft.CodeAnalysis.CodeActions
     /// </summary>
     public class DocumentNavigationOperation : CodeActionOperation
     {
-        private readonly DocumentId _documentId;
-        private readonly int _position;
+        internal DocumentId DocumentId { get; }
+        internal int Position { get; }
 
-        public DocumentNavigationOperation(DocumentId documentId, int position = 0)
+        public DocumentNavigationOperation(DocumentId documentId!!, int position = 0)
```
[the remark this answers is not in the document]
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/Common/DocumentNavigationOperation.cs]: had to remove as well, was getting formatting warnings :'(

**Pull request merged**